### ISSUE TITLE
Seems text_match() function doesn't support search with  '#'  character

### ISSUE THREAD

**Issue by Betternan** [apache/pinot#12262]
The data in Pinot table "x1" looks like:
ID and tags column are text type
ID | tags
--|-----
1  | ABC
2 | #ABC
3 | ABC, DEF
4 | ABC, CDE

When I use following SQL:

select count(distinct (ID)) from x1 where text_match(tags, '#ABC');

The query will return 4 as result, but I think the expect result would be 1, since only one record match with '#ABC'.

I tried with following text_match write syntax, but it seems the same result.
text_match(tags, '"#ABC"')
text_match(tags, '\#ABC')
text_match(tags, '\#ABC')

When I use following syntax, the query returns 0.
text_match(tags, '/#ABC.*/')

Is there anybody meet this case? Or any solution for this? Waiting for your reply. Thanks.

**Comment by Betternan** [comment]
Hi @walterddr ,  I saw you marked this case as a bug, so do you have any clue for this？

**Comment by Betternan** [comment]
Or is there any workaround to hit this match?

**Comment by walterddr** [comment]
i haven't got a chance to take a look at it. can you share the minimum repro? 
e.g. table config / schema / possibly csv data i can use to ingest?

**Comment by Betternan** [comment]
@Jackie-Jiang , yes, the "tags" column has configured with text index. I tried the same syntax with you before I propose this issue, but seems it doesn't work.

**Comment by Jackie-Jiang** [comment]
I saw this similar [issue](https://discuss.elastic.co/t/lucene-query-for-string-containing-hash-symbol/316429) posted for Elastic search, but no response..

**Comment by Betternan** [comment]
@Jackie-Jiang Seems no response for this kinds of problem. But this is major case I met, do you have any idea how can I solve this problem?

**Comment by Jackie-Jiang** [comment]
I find something [here](https://solr-user.lucene.apache.narkive.com/X6ReA2cA/search-for-the-single-hash-character-never-returns-results) that might help: `text_match(tags, '[# TO #]ABC')`
Again, I'm not a Lucene expert, so not sure how it works internally

**Comment by kkrugler** [comment]
Pinot uses Lucene's StandardAnalyzer to convert text into "tokens", which are then used for matching against queries. The StandardAnalyzer removes most punctuation, so the behavior you're seeing might be caused by the `#` characters being removed during indexing.

**Comment by Betternan** [comment]
@kkrugler , so you mean Pinot might need to customize the StandardAnalyzer to suit "#" search character, right?

**Comment by kkrugler** [comment]
Correct - currently I think the analysis chain being used is hard coded. There are some issues filed about improving this, but I haven't seen much activity. See https://github.com/apache/pinot/issues/10374

**Comment by itschrispeck** [comment]
You can configure the Analyzer used to make this searchable (https://github.com/apache/pinot/pull/12027, https://github.com/apache/pinot/pull/12101)
